src/net/tls.cc

```diff
@@ -1533,25 +1533,110 @@ class session : public enable_lw_shared_from_this<session> {
         result_t dn = extract_dn_information();
         return make_ready_future<result_t>(std::move(dn));
     }
+    future<std::vector<subject_alt_name>> get_alt_name_information(std::unordered_set<subject_alt_name_type> types) {
+        using result_t = std::vector<subject_alt_name>;
+
+        if (_error) {
+            return make_exception_future<result_t>(_error);
+        }
+        if (_shutdown) {
+            return make_exception_future<result_t>(std::system_error(ENOTCONN, std::system_category()));
+        }
+        if (!_connected) {
+            return handshake().then([this, types = std::move(types)]() mutable {
+               return get_alt_name_information(std::move(types));
+            });
+        }
+
+        auto gris = extract_dn_information();
```

Comment:
gris?

Reply:
Oops. Debug line that made it into commit...

Reply:
Removed